indico/modules/events/layout/models/menu.py

```diff
@@ -125,6 +125,8 @@ def is_orphaned(self):
     def is_visible(self):
         if not self.is_enabled:
             return False
+        if self.registered_only and not session.user.is_registered_in_event(self.event):
```

Comment:
needs to check for the `not session.user` case since it may be None.

Alternatively (maybe that keeps the code saner here), you could replace `User.is_registered_in_event` with `Event.is_user_registered`:

```python
def is_user_registered(self, user):
    """Check whether the user is registered in the event.
    
    This takes both unpaid and complete registrations into account.
    """
    from indico.modules.events.registration.models.registrations import Registration, RegistrationState
    if user is None:
        return False
    return (Registration.query.with_parent(self)
            .join(Registration.registration_form)
            .filter(Registration.user == user,
                    Registration.state.in_([RegistrationState.unpaid, RegistrationState.complete]),
                    ~Registration.is_deleted,
                    ~RegistrationForm.is_deleted)
            .has_rows())
```